test/projectile-test.el

```diff
@@ -1163,8 +1163,23 @@ Just delegates OPERATION and ARGS for all operations except for`shell-command`'.
            "spec/foo/foo.service.spec.js"
            "spec/bar/bar.service.spec.js")
           (:test-suffix ".spec" :test-dir "spec/" :src-dir "source/")
-        (expect (projectile--find-matching-test "source/foo/foo.service.js") :to-equal '("spec/foo/foo.service.spec.js"))
-        (expect (projectile--find-matching-file "spec/bar/bar.service.spec.js") :to-equal '("source/bar/bar.service.js")))))
+        (expect (projectile--find-matching-test
+                 "project/source/foo/foo.service.js")
+                :to-equal '("spec/foo/foo.service.spec.js"))
+        (expect (projectile--find-matching-file
+                 "project/spec/bar/bar.service.spec.js")
+                :to-equal '("source/bar/bar.service.js")))))
+
+  (it "finds matching test with dirs and inexistent test file"
+    (projectile-test-with-sandbox
+      (projectile-test-with-files-using-custom-project
+          ("project/src/main/scala/bar/package.scala"
+           "project/src/main/scala/foo/package.scala"
+           "project/src/test/scala/foo/packageSpec.scala")
+          (:test-suffix "Spec" :test-dir "test" :src-dir "main")
+        (expect (projectile--find-matching-test
+                 "project/src/main/scala/bar/package.scala")
+                :to-equal '("src/test/scala/bar/packageSpec.scala")))))
```

Comment:
This would fail before this change, since https://github.com/bbatsov/projectile/blob/2bb7ec28b1275bbce7cac743ee9e7b2cf41c5bbd/projectile.el#L3574 would take precedence over using the `src-dir`/`test-dir` properties.